wrappers/python/examples/llama_training.py

```diff
@@ -103,114 +96,234 @@
+        nntile.starpu.restrict_cpu()
+    time1 = time.time() - time0
+    print("StarPU + NNTile + MPI init in {} seconds".format(time1))
+    next_tag = 0
+
+    time0 = time.time()
+    if args.n_head_tile == -1:
+        args.n_head_tile = model_torch.config.num_attention_heads
+    if args.hidden_size_tile == -1:
+        args.hidden_size_tile = model_torch.config.hidden_size
+    if args.intermediate_size_tile == -1:
+        args.intermediate_size_tile = model_torch.config.intermediate_size
+
+    llama_config_nntile = LlamaConfigNNTile(
+        vocab_size=model_torch.vocab_size,
+        vocab_embed_dim_tile=model_torch.config.hidden_size,
+        hidden_size=model_torch.config.hidden_size,
+        hidden_size_tile=args.hidden_size_tile,
+        max_position_embeddings=model_torch.config.max_position_embeddings,
+        num_hidden_layers=model_torch.config.num_hidden_layers,
+        rms_norm_eps=model_torch.config.rms_norm_eps,
+        n_attention_head=model_torch.config.num_attention_heads,
+        num_key_value_heads=model_torch.config.num_key_value_heads,
+        intermediate_size=model_torch.config.intermediate_size,
+        intermediate_size_tile=args.intermediate_size_tile,
+        n_head_tile=args.n_head_tile,
+        dtype=args.nntile_dtype
+        )
+
+    print(llama_config_nntile)
+
+    single_batch_pos_ids = np.arange(args.seq_len).reshape(1, args.seq_len)
+    pos_ids = np.repeat(single_batch_pos_ids, args.minibatch_size, axis=0)
+
+    mask = np.array(np.triu(np.ones((args.seq_len, args.seq_len))),
+                        dtype=bool, order="F")
+    llama_nntile, next_tag = Llama_nntile.from_torch(model_torch,
+                                                    args.minibatch_size,
+                                                    args.minibatch_size_tile,
+                                                    args.seq_len,
+                                                    args.seq_len_tile,
+                                                    pos_ids,
+                                                    mask,
+                                                    llama_config_nntile,
+                                                    next_tag)
+    time1 = time.time() - time0
+    print("Converting PyTorch model to NNTile",
+          "requires {} seconds".format(time1))
+    del model_torch
+
+if args.nntile_nepochs > 0:
+    from pathlib import Path
+
+    import numpy as np
+    from datasets import load_dataset
+    if args.dataset == "tinystories":
+        if args.dataset_file == args.dataset + "_train.bin":
+            train_data = np.memmap(Path(args.dataset_path) /
+                                    args.dataset_file,
+                                   dtype=np.uint16, mode='r')
+            train_tokens_raw = np.array(train_data, order='F', dtype=np.int64)
+            del train_data
+        elif args.dataset_file == "":
+            train_dataset = load_dataset("roneneldan/TinyStories",
```

Comment:
Since we aim this script to digest inputs as `train.bin`, this check might become unneseccary